lib/pbench/test/unit/server/test_requests.py

```diff
@@ -4,18 +4,11 @@
 import pytest
 import re
 import requests
-import responses
 
 from pathlib import Path
 from werkzeug.utils import secure_filename
 
 
-@pytest.fixture
-def mocked_responses():
-    with responses.RequestsMock() as rsps:
-        yield rsps
```

Comment:
I tried to change my mocks to use this model, but I wasn't able to find any way to get my exception tests to pass. My guess, although I couldn't find enough context to prove it, is that the `RequestsMock` overloading of `body` instead of having a separated `exc` argument, was interfering with the payload in the tested API's `post`, and generating type errors.

I couldn't find any way to fix this, so instead of converted your one mock to use the `requests_mock` package, instead. In theory, I think we could therefore drop `responses` from the `pbench-devel` container build, but I decided not to be that aggressive. There doesn't appear to be an RPM for `requests_mock`, either, so we'll have to rely on the pip install inside the container.